### PR TITLE
Feature/add more election results data

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Closes https://github.com/DemocracyClub/WhoCanIVoteFor/issues/935 and https://github.com/DemocracyClub/WhoCanIVoteFor/issues/1663
Dependent on https://github.com/DemocracyClub/yournextrepresentative/pull/2174

This work adds election results for past elections, when results are available.

- [x] Add turnout, ballot papers issues, electorate and spoilt ballots to `Election` model
- [x] Display ^^ results data in a new table on the ballot template as shown below, only when they exist
- [x] Add Welsh translations
- [x] Import this data as part of the `import_ballots` process
- [x] Add election results to YNR ballot serializer
- [x] Switches candidate lists to a single column

To test:
- run migrations
- run the ballot importer for recent elections with `python manage.py import_ballots --current`
- visit a [recently past election](http://localhost:8000/elections/local.st-albans.marshalswick-east-jersey-farm.by.2023-08-17/marshalswick-east-jersey-farm/) with results to view a new table similar to the one shown below

Screenshot 2023-09-19 at 4 41 03 PM

Screenshot 2023-09-19 at 4 41 19 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205303544790826

### CODE REVIEW

**Comment by VirginiaDooley** [comment]
@pmk01 Would you prefer the table not appear at all if results are not available? Or should it read, as is the case currently, eg "Electorate not available"?

**Comment by VirginiaDooley** [comment]
This works well with current ballots, but for a full import, I hit an issue with formatting.  For example: https://candidates.democracyclub.org.uk/api/next/results/local.reading.norcot.by.2023-08-03/ which points to an issue with the way I've formatted it in the serializer or the way it's arriving via `resultsbot`

**Comment by pmk01** [comment]
> @pmk01 Would you prefer the table not appear at all if results are not available? Or should it read, as is the case currently, eg "Electorate not available"?

We should not show the table until we have some results.

It will be common for us to lack some numbers. It's often the case that councils only publish turnout percentage and nothing else, for example. We could just leave the field blank if we don't have the number? I think this is preferable to "[thing] not available".

**Comment by VirginiaDooley** [comment]
> > @pmk01 Would you prefer the table not appear at all if results are not available? Or should it read, as is the case currently, eg "Electorate not available"?
> 
> We should not show the table until we have some results.
> 
> It will be common for us to lack some numbers. It's often the case that councils only publish turnout percentage and nothing else, for example. We could just leave the field blank if we don't have the number? I think this is preferable to "[thing] not available".

I think we have to have something in the box if a data point isn't available - if nothing else to maintain the formatting of the table, is "None" ok?

**Comment by pmk01** [comment]
Sorry, no, 'none' wouldn't work becuase turnout: "none" doesn't make any sense. If we have nothing at all, does the column disappear? I am fine with that if the case.

**Comment by symroe** [comment]
Sorry to throw a spanner in the works late on, but would it be better for the table to be flopped 90°?

So we have

```
a: 1
b: 2
c: 3
```
rather than 

```
a    b    c
1    2    3

```

Thinking about mobile interfaces mostly,

**Comment by VirginiaDooley** [comment]
> Sorry to throw a spanner in the works late on, but would it be better for the table to be flopped 90°?
> 
> So we have
> 
> ```
> a: 1
> b: 2
> c: 3
> ```
> 
> rather than
> 
> ```
> a    b    c
> 1    2    3
> ```
> 
> Thinking about mobile interfaces mostly,

Agree. Better? Or both and make it responsive based on device? 

Mobile
Screenshot 2023-09-19 at 12 35 57 PM

Desktop 
Screenshot 2023-09-19 at 12 35 32 PM

File: wcivf/assets/scss/style.scss [hunk 1]
```diff
@@ -156,3 +156,10 @@ input[type=radio] {
 .ds-status-message p {
   color: $white;
 }
+
+#results-table {
```
Comment (symroe) [inline wcivf/assets/scss/style.scss:160]: I think this can go now?

**Review by symroe — APPROVED** [review]
One comment and I think the commits need cleaning up, but this is good to go.

One thing I've not tested is the `has_results` function in reality. It is possible that we have candidate results but no values in the listed fields?

File: wcivf/assets/scss/style.scss (hunk 1 above)
Comment (VirginiaDooley) [reply wcivf/assets/scss/style.scss:160]: Still using this.

File: wcivf/apps/elections/models.py [hunk 2]
```diff
@@ -239,6 +243,19 @@ def pluralized_division_name(self):
...
+    @property
+    def has_results(self):
+        """
+        Returns a boolean for if the election has results
+        """
+        return bool(
+            self.spoilt_ballots
+            or self.ballot_papers_issued
+            or self.turnout
+            or self.electorate
```
Comment (VirginiaDooley) [inline wcivf/apps/elections/models.py:255]: I think this is what you suggested? @symroe

Comment (symroe) [reply wcivf/apps/elections/models.py:255]: I think the last clause can also be a n `or`?

Comment (VirginiaDooley) [reply wcivf/apps/elections/models.py:255]: I'm not sure about that; I didn't think we wanted to display the table unless at least one of the above fields are not `None`.

Comment (VirginiaDooley) [reply wcivf/apps/elections/models.py:255]: I've renamed this for clarity `has_election_results`.

**Pull request merged**